// gitbook-plugin-mydemosamples/index.js
var ncp = require('ncp').ncp;
var fs = require('fs');
var path = require('path');
ncp.limit = 16;

module.exports = {
    book: {
        assets: "./book",
        js: [
            "test.js"
        ],
        css: [
            "test.css"
        ]
    },
    hooks: {
        init: function() {
            console.log("init5151651!");
        },
        finish: function() {
            console.log("finish");
            console.log("him!");
            ncp("./ol3/", "./_book/ol3/", function (err) {
                if (err) {
                    return console.error(err);
                }
                console.log('done!');
            });
            var p = './';
            fs.readdir(p, function (err, files) {
                if (err) {
                    throw err;
                }

                files.map(function (file) {
                    return path.join(p, file);
                }).filter(function (file) {
                    return fs.statSync(file).isFile();
                }).forEach(function (file) {
                    if (path.extname(file) === '.html') {
                        console.log("%s (%s)", file, path.extname(file));
                        fs.createReadStream(file).pipe(fs.createWriteStream( '_book/' + file));
                    }
                });
            });
        }
    }
};
